controllers/echoserver_controller.go

```diff
@@ -97,6 +97,8 @@ func (r *EchoServerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	// not working yet
 	isMarkedToBeDeleted := echoServer.DeletionTimestamp != nil
 	if isMarkedToBeDeleted {
+		echoServer.Status.State = "Deleting"
```

Comment:
We should get in the habit of using `constants` here.  

`// Common event reasons
const (
	Created string = "Created"
	Failed  string = "Failed"
	Deleting  string = "Deleting"
```
suggestion
		echoServer.Status.State = Deleting
```
)`